ngraph/core/include/ngraph/descriptor/tensor.hpp

```diff
@@ -53,7 +55,9 @@ namespace ngraph
             void set_lower_value(const HostTensorPtr& value);
             /// \brief sets upper bound value description
             void set_upper_value(const HostTensorPtr& value);
-            /// \brief unsets bound value descriptions
+            /// \brief sets value label description
+            void set_value_label(const TensorLabel& value_label);
```

Comment:
Why do we need to have value labels? Do we want to standardise labels? Or can it have any abstract names?

Reply:
Labels are needed for the places in the Function where shapes turn into value -- like ShapeOf nodes. Please see the example:
```
       SomeOp
out shape: [SpecialDim:1, 3, 224, 224]
            |
          \  /
      ShapeOf
out shape: [4]
out value: [1, 3, 224, 224]
out value label: ["SpecialDim", "", "", ""]
            |
            |  in:1
          \  /
     Reshape
out shape: [SpecialDim:1, 3, 224, 224]
```
There is no point in standardizing them -- these are any string that we or user could think of.